### PR TITLE
ROX-19968: Add permission checks for watched images controls

### PR DESCRIPTION
**Pull request by dvail**
## Description

As titled. For simplicity and consistency with VM 1.0, watched image functionality will be restricted to users with "READ_WRITE_ACCESS" on the "WatchedImage" resource.

"Hide whitespace" for easier reviewing.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Visit the Workload CVE list page, on the "Image" tab, with less than "READ_WRITE" permissions.
image

Revisit the page with full permissions.
image

### CODE REVIEW

**Comment by dvail** [comment]
Current dependencies on/for this PR:
* master
  * **PR #8013** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/8013" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/8013?utm_source=stack-comment).

File: ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx [hunk 1]
```diff
@@ -37,6 +38,8 @@ const emptyStorage: VulnMgmtLocalStorage = {
 
 function WorkloadCvesOverviewPage() {
     const apolloClient = useApolloClient();
+    const { hasReadWriteAccess } = usePermissions();
+    const canWriteWatchedImages = hasReadWriteAccess('WatchedImage');
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx:42]: An alternative to consider, mostly taste although somewhat for Find in Files:

```ts
const hasWriteAccessForWatchedImage = hasReadWriteAccess('WatchedImage');
```

Having read or written that so many times recently, `hasWriteAccess` function name would be clearer and worth discussion whether to replace all occurrences in 4.4A sprint.

File: ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx [hunk 2]
```diff
@@ -80,17 +83,19 @@ function WorkloadCvesOverviewPage() {
...
-                        onClick={() => {
-                            setDefaultWatchedImageName('');
-                            watchedImagesModalToggle.openSelect();
-                        }}
-                    >
-                        Manage watched images
-                    </Button>
-                </FlexItem>
+                {canWriteWatchedImages && (
+                    <FlexItem className="pf-u-pr-lg">
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx:87]: Outside of these changes, the utility class caught my eye.

The alignment of the button with the inner white section below it makes total sense, but I was curious why why `PageSection` has `'noPadding'` and then it seems like its descendants specify the padding.

File: ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx [hunk 3]
```diff
@@ -80,17 +83,19 @@ function WorkloadCvesOverviewPage() {
...
+                        <Button
+                            variant="secondary"
+                            onClick={() => {
+                                setDefaultWatchedImageName('');
+                                watchedImagesModalToggle.openSelect();
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx:92]: Aha, here is an example where we talked about possibility of a more generic hook.

**Review by pedrottimark — APPROVED** [review]

File: ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx (hunk 2 above)
Comment (dvail) [reply ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx]: Good question - I'm not sure what the original reason was but it looks like it is no longer valid. I removed the `noPadding` prop and the utility classes and the layout was identical. (At least on larger screen sizes; the padding is less on tablet and lower sized devices, but is consistent with the PatternFly defaults.)

**Pull request merged**